src/services/FirebaseService.ts

```diff
@@ -0,0 +1,75 @@
+import 'firebase/auth';
+
+import firebase from 'firebase/app';
+
+const config = {
+  apiKey: process.env.REACT_APP_FIREBASE_API_KEY,
+  authDomain: process.env.REACT_APP_FIREBASE_AUTH_DOMAIN,
+  projectId: process.env.REACT_APP_FIREBASE_PROJECT_ID,
+  storageBucket: process.env.REACT_APP_FIREBASE_STORAGE_BUCKET,
+  messagingSenderId: process.env.REACT_APP_FIREBASE_MESSAGING_SENDER_ID,
+  appId: process.env.REACT_APP_FIREBASE_APP_ID,
+};
+
+class Firebase {
+  auth: firebase.auth.Auth;
+
+  constructor() {
+    if (!firebase.apps.length) {
+      firebase.initializeApp(config);
+    } else {
+      firebase.app(); // if already initialized, use that one
+    }
+    this.auth = firebase.auth();
+  }
+
+  // *** Auth API ***
+  // Creates a user and signs the user in
+  doCreateUserWithEmailAndPassword = (email: string, password: string) => {
```

Comment:
Minor thing I forgot to mention last time, let's use regular function declarations instead of arrow functions for functions inside classes so:
```suggestion
  doCreateUserWithEmailAndPassword(email: string, password: string) {
```

Reply:
Also, just merged in a change to the linter that forces us to add types to return values of functions so you'll probably have to add that!